src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -534,7 +533,7 @@ private AbstractFrame getFieldWithoutThis(DetailAST ast, int parentType) {
      */
     private static boolean isInCompactConstructor(DetailAST ast) {
         boolean isInCompactCtor = false;
-        DetailAST parent = ast.getParent();
```

Comment:
This is only called through `processIdent`, so the AST is always an `IDENT`.

This is very similar to other things we have seen.

The loop below looks for some criteria and until it finds it it goes up the parentage. Well this mutation is removing going up the first parent. The only way we can say its needed is if the AST is what we are looking for, `COMPACT_CTOR_DEF` and we won't since the AST is always `IDENT`.

The AST is never `null` as seen from the original code, otherwise it would be an NPE on getting the parent.